core/base/planarGraphLayout/MergeTreeVisualization.h

```diff
@@ -0,0 +1,930 @@
+///
+/// Visualization module for merge trees.
+///
+
+#ifndef _MERGETREEVISUALIZATION_H
+#define _MERGETREEVISUALIZATION_H
+
+#pragma once
+
+#include <FTMTree.h>
+
+#include <stack>
+
+using namespace ttk;
+using namespace ftm;
+
+class MergeTreeVisualization : virtual public Debug {
+protected:
+  // Visualization parameters
+  bool PlanarLayout = false;
+  bool BranchDecompositionPlanarLayout = false;
+  double BranchSpacing = 1.;
+  bool RescaleTreesIndividually = false;
+  double ImportantPairs = 50.; // important pairs threshold
+  double ImportantPairsSpacing = 1.;
+  double NonImportantPairsSpacing = 1.;
+  double NonImportantPairsProximity = 0.05;
+
+public:
+  MergeTreeVisualization(){};
+  ~MergeTreeVisualization(){};
+
+  // ==========================================================================
+  // Getter / Setter
+  // ==========================================================================
+  // Visualization parameters
+  void setPlanarLayout(bool b) {
+    PlanarLayout = b;
+  }
+  void setBranchDecompositionPlanarLayout(bool b) {
+    BranchDecompositionPlanarLayout = b;
+  }
+  void setBranchSpacing(double d) {
+    BranchSpacing = d;
+  }
+  void setRescaleTreesIndividually(bool b) {
+    RescaleTreesIndividually = b;
+  }
+  void setImportantPairs(double d) {
+    ImportantPairs = d;
+  }
+  void setImportantPairsSpacing(double d) {
+    ImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsSpacing(double d) {
+    NonImportantPairsSpacing = d;
+  }
+  void setNonImportantPairsProximity(double d) {
+    NonImportantPairsProximity = d;
+  }
+
+  // ==========================================================================
+  // Planar Layout
+  // ==========================================================================
+  // TODO manage multi pers pairs
+  template <class dataType>
+  void treePlanarLayoutBDImpl(
+    FTMTree_MT *tree,
+    std::vector<float> &retVec,
+    std::vector<LongSimplexId> &treeSimplexId,
+    std::vector<idNode> &branching,
+    std::vector<std::vector<ftm::idNode>> &nodeBranching) {
+    idNode treeRoot = tree->getRoot();
+    idNode treeRootOrigin = tree->getNode(treeRoot)->getOrigin();
+    float rootY = retVec[treeSimplexId[treeRoot] * 2 + 1];
+    float rootOriginY = retVec[treeSimplexId[treeRootOrigin] * 2 + 1];
+    float rootYmin = std::min(rootY, rootOriginY);
+    float rootYmax = std::max(rootY, rootOriginY);
+    dataType rootPers = tree->getNodePersistence<dataType>(treeRoot);
+    std::vector<std::tuple<float, float>> allNodeSpanX(
+      tree->getNumberOfNodes());
+    std::vector<std::tuple<float, float>> allNodeImportantSpanX(
+      tree->getNumberOfNodes());
+
+    // Compute gap
+    float nonImportantPairsGap
+      = (rootYmax - rootYmin) * 0.05 * NonImportantPairsSpacing;
+    float importantPairsGap
+      = std::max(nonImportantPairsGap, (float)ImportantPairsSpacing);
+
+    // Some functions
+    auto comp = [&](const ftm::idNode a, const ftm::idNode b) {
```

Comment:
I would give explicit name with Lower in it

Reply:
Done in the next commit!